VS Code extension

sungshik opened the pull request:
This PR adds a subdirectory to use Bird in a standalone VS Code extension. It is essentially a copy/paste of [this example](https://github.com/usethesource/rascal-language-servers/tree/main/examples/pico-dsl-extension), but with minimal changes to use Bird.

DavyLandman commented on README.md:
```diff
@@ -18,9 +18,10 @@ We provide an IDE that allows users to edit and compile Bird and Nescio descript
+- `vscode-extension`: The VS Code extension for Bird. It leverages the Bird language server to offer a full IDE experience.
 
-In order to build them all, clone this project and execute `mvn install` on the root subdirectory. Notice that it is necessary first to build the [Nescio project](https://github.com/SWAT-engineering/nescio), since the Bird components related to the integration with Nescio depend on it.
+To build `bird-core` and `bird-ide`, clone this project and execute `mvn install` on the root subdirectory. Notice that it is necessary first to build the [Nescio project](https://github.com/SWAT-engineering/nescio), since the Bird components related to the integration with Nescio depend on it. To build a VSIX that can be installed in VS Code, run `npm i && npx vsce package` in the `vscode-extens ...
```
I think this paragraph is outdated.

we need to do `mvn install` on nest, but the rest is just a local build (and `nescio` we haven't gotten back yet).

DavyLandman commented on bird.code-workspace:
```diff
@@ -23,6 +23,9 @@
 			"path": "."
+		},
+		{
+			"path": "vscode-extension"
```
shall we rename it to `bird-extension` to align the project names?

DavyLandman commented on vscode-extension/.vscode/tasks.json:
```diff
@@ -0,0 +1,40 @@
+// for the documentation about the tasks.json format
+{
+	"version": "2.0.0",
+	"tasks": [
```
TODO for the example in rascal-language-servers: rewrite the example to use esbuild instead of webpack.

sungshik replied:
Created an [issue](https://github.com/usethesource/rascal-language-servers/issues/614)

DavyLandman commented on vscode-extension/package.json:
```diff
@@ -0,0 +1,61 @@
+    "webpack-cli": "^4.10.0"
+  },
+  "dependencies": {
+    "@usethesource/rascal-vscode-dsl-lsp-server": "0.12.0"
```
let's use 0.12.2 to be up to date with the latest.

sungshik commented:
Updated the categories (with help from Rodin), regenerated the grammar (and made a few manual improvements), and fixed the other small things. Merging now.
